pkg/plugins/shell/condition.go

```diff
@@ -0,0 +1,46 @@
+package shell
+
+import (
+	"fmt"
+
+	"github.com/sirupsen/logrus"
+	"github.com/updatecli/updatecli/pkg/core/result"
+	"github.com/updatecli/updatecli/pkg/core/scm"
+)
+
+// Condition tests if the provided command (concatenated with the source) is executed with success
+func (s *Shell) Condition(source string) (bool, error) {
+	return s.condition(source, "")
+}
+
+// ConditionFromSCM tests if the provided command (concatenated with the source) is executed with success from the SCM root directory
+func (s *Shell) ConditionFromSCM(source string, scm scm.Scm) (bool, error) {
+	return s.condition(source, scm.GetDirectory())
+}
+
+func (s *Shell) condition(source, workingDir string) (bool, error) {
+	customCommand := s.spec.Command
+	if customCommand == "" {
+		return false, fmt.Errorf(ErrEmptyCommand)
+	}
+	if source != "" {
+		customCommand += " " + source
+	}
+	cmdResult, err := s.executor.ExecuteCommand(command{
+		Cmd: customCommand,
+		Dir: workingDir,
+	})
+
+	if err != nil {
+		return false, err
+	}
+
+	if cmdResult.ExitCode != 0 {
+		logrus.Infof("%v The shell 🐚 command %q failed to validate the condition:\nexit code was %q\nstderr=\n%v\nstdout=\n%v\n.", result.FAILURE, customCommand, cmdResult.ExitCode, cmdResult.Stderr, cmdResult.Stdout)
```

Comment:
Not sure if it is relevant, but logrus allows to do structured logging, ie adding attributes to the log line and let the logger handle the formatting, instead of doing it by yourself.

The upside here is that you can change the formatting of the whole log line to `JSON` for example and still have those attributes parse-able.

For (incomplete) example....

```suggestion
		 logrus.WithFields(log.Fields{
			"exit_code": cmdResult.ExitCode,
			"stdout": cmdResult.Stdout,
			"stderr": cmdResult.Stderr, 
  		}).Info("Unable to execute command...")
```